lua/vgit/features/LiveGutter.lua

```diff
@@ -22,6 +22,27 @@ LiveGutter.hide = loop.async(function(_, buffer)
   buffer:sign_unplace()
 end)
 
+LiveGutter.store_buffer_vars = loop.async(function(_, buffer, hunks)
```

Comment:
Each buf number is encapsulated inside a [`Buffer`](https://github.com/tanvirtin/vgit.nvim/blob/main/lua/vgit/core/Buffer.lua#L13) object in VGit, which may have a composition relationship with [`GitObject`](https://github.com/tanvirtin/vgit.nvim/blob/main/lua/vgit/core/Buffer.lua#L27). Since we are generating status for a buffer I think this logic is a good candidate to be inside [`GitObject`](https://github.com/tanvirtin/vgit.nvim/blob/main/lua/vgit/git/GitObject.lua#L12) :)

e.g
```lua
buffer.git_object:generate_diff_status()
```

Reply:
Makes sense, do you also think the call to `generate_diff_status` should stay inside `LiveGutter` or move to the `Buffer` object?

Maybe better save this discussion as well for #220 :-)